src/drivers/rog_ally/driver.rs

```diff
@@ -0,0 +1,69 @@
+use std::{
+    error::Error,
+    ffi::{CString, OsStr},
+};
+
+use udev::Device;
+
+use crate::udev::device::{AttributeSetter, UdevDevice};
+
+// Hardware ID's
+const ALLY_PID: u16 = 0x1abe;
+const ALLYX_PID: u16 = 0x1b4c;
+pub const PIDS: [u16; 2] = [ALLY_PID, ALLYX_PID];
+pub const VID: u16 = 0x0b05;
+
+pub struct Driver {
+    _device: UdevDevice,
+}
+
+impl Driver {
+    pub fn new(udevice: UdevDevice) -> Result<Self, Box<dyn Error + Send + Sync>> {
+        let vid = udevice.id_vendor();
+        let pid = udevice.id_product();
+        if VID != vid || !PIDS.contains(&pid) {
+            return Err(format!("'{}' is not an ROG Ally controller", udevice.devnode()).into());
+        }
+
+        // Set the controller buttons to the correct values at startup
+        let device = udevice.get_device()?;
+        let Some(mut parent) = device.parent() else {
+            return Err("Failed to get device parent".into());
+        };
+        let Some(driver) = parent.driver() else {
+            return Err("Failed to identify device driver".into());
+        };
+        if driver.to_str().unwrap() == "asus" {
+            set_attribute(device.clone(), "btn_m1/remap", "kb_f15")?;
+            set_attribute(device.clone(), "btn_m2/remap", "kb_f14")?;
+            set_attribute(device, "gamepad_mode", "1")?;
```

Comment:
So all this driver does is set udev attributes and rely on the evdev device? This works for now, but maybe we should add a `TODO` for adding a new concept to address cases like these so we're not wasting CPU cycles polling an empty implementation.

Reply:
Yes, we'll need it for OXP devices as well to take over the turbo button